test/add-member.test.js

```diff
@@ -42,3 +43,75 @@ test('get added to a group', async (t) => {
     )
   )
 })
+
+test('add member', async (t) => {
+  const kaitiaki = Testbot()
+  const newPerson = Testbot()
+  kaitiaki.tribes2.start()
+  newPerson.tribes2.start()
+
+  try {
+    const group = await kaitiaki.tribes2.create()
+    t.true(group.id, 'creates group')
+
+    const authorIds = [newPerson.id, ssbKeys.generate().id]
+
+    let invite = await kaitiaki.tribes2.addMembers(group.id, authorIds, {
+      text: 'welcome friends',
+    })
+
+    invite = await p(kaitiaki.db.get)(invite.key)
+
+    const expected = {
+      type: 'group/add-member',
+      version: 'v1',
+      groupKey: group.secret.toString('base64'),
+      root: group.root,
+
+      text: 'welcome friends',
+      recps: [group.id, ...authorIds],
+
+      tangles: {
+        group: { root: group.root },
+        members: { root: group.root, previous: [group.root] },
+      },
+    }
+    // we don't know the key of the last message, that was the admin adding themselves
+    expected.tangles.group.previous = invite.content.tangles.group.previous
+    t.deepEqual(invite.content, expected, 'kaitiaki sent invite')
+
+    /* kaitiaki posts to group, new person can read */
+    const greetingContent = {
+      type: 'post',
+      text: 'Welcome new person!',
+      recps: [group.id],
+    }
+    const { key: greetingKey } = await kaitiaki.tribes2.publish(greetingContent)
+    await replicate(kaitiaki, newPerson)
+    const greetingMsg = await p(newPerson.db.getMsg)(greetingKey)
+    t.deepEqual(
+      greetingMsg.value.content,
+      greetingContent,
+      'new person can read group content'
+    )
+
+    /* new person posts to group, kaitiaki can read */
+    const replyContent = {
+      type: 'post',
+      text: 'Thank you kaitiaki',
+      recps: [group.id],
+    }
+    const { key: replyKey } = await newPerson.tribes2.publish(replyContent)
+    await replicate(newPerson, kaitiaki)
+    const replyMsg = await p(kaitiaki.db.getMsg)(replyKey)
+    t.deepEqual(
+      replyMsg.value.content,
+      replyContent,
+      'kaitiaki can read things from new person'
+    )
+  } catch (err) {
+    t.fail(err)
+  }
+
+  kaitiaki.close(true, () => newPerson.close(true))
```

Comment:
This will hit the "async test" corner case. We have to do this instead:

```suggestion
  await new Promise((resolve) => {
    kaitiaki.close(true, () => newPerson.close(true, resolve))
  })
```